File: c/stack/TestStack.c
#include "Stack.h"
#include <stdio.h>

int test_stack() {
    printf("START - test stack\n");
    Stack *my_stack = stack_init();
    stack_insert(my_stack, 2);
    stack_insert(my_stack, 6);
    stack_insert(my_stack, 7);
    stack_print(my_stack);
    stack_insert(my_stack, 0);
    stack_print(my_stack);
    int result = stack_pop(my_stack);
    if (result != 0) {
        printf("ERROR: invlaid value returned by stack_pop\n");
        return 1;
    }
    printf("END - test stack\n");
    return 0;
}

File: typescript/Stack/TestStack.ts
import { Stack } from './Stack';

const testStack = () => {
  const stack = new Stack<number>();

  stack.push(1);
  stack.push(2);
  stack.print();
  console.log(stack.pop());
  console.log(stack.isEmpty());
  console.log(stack.pop());
  console.log(stack.isEmpty());
};

testStack();